File: app/src/main/java/tr/name/fatihdogan/books/dependecy/module/PreferencesModule.java
package tr.name.fatihdogan.books.dependecy.module;

import android.app.Application;
import android.content.SharedPreferences;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class PreferencesModule {

    private static final String PREFERENCES_NAME = "MyPref";

    @Provides
    @Singleton
    SharedPreferences provideSharedPreferences(Application application) {
        return application.getSharedPreferences(PREFERENCES_NAME, 0);
    }

}

File: app/src/main/java/com/fthdgn/books/dependecy/module/PreferencesModule.kt
package com.fthdgn.books.dependecy.module

import android.app.Application
import android.content.SharedPreferences
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class PreferencesModule {
    @Provides
    @Singleton
    internal fun provideSharedPreferences(application: Application): SharedPreferences =
            application.getSharedPreferences("MyPref", 0)
}
